api/types.go

```diff
@@ -166,12 +166,12 @@ type ToolFunction struct {
 	Name        string `json:"name"`
 	Description string `json:"description"`
 	Parameters  struct {
-		Type       string   `json:"type"`
-		Required   []string `json:"required"`
+		Type       interface{} `json:"type"`
```

Comment:
Spent a bit of time playing around with this. I tried to define a type with custom marshaling but that seems to have unintended consequences with tests around tool calling and it seemed like a bit of a rabbit hole to get into.

Thoughts are welcomed here but also want to push to get the Tool passing stuff out